dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/impl/MetricRegistryImpl.java

```diff
@@ -433,7 +433,7 @@ public void removeMatching(MetricFilter filter) {
     }
 
     @SuppressWarnings("unchecked")
-    private <T extends Metric> T getOrAdd(Metadata metadata, MetricBuilder<T> builder) {
+    private synchronized <T extends Metric> T getOrAdd(Metadata metadata, MetricBuilder<T> builder) {
```

Comment:
I think we don't need `synchronized` here.  `register()` method will throw an `IllegalArgumentException` when it detects duplication.  I think we just need to add `@FFDCIgnore(IllegalArgumentException.class)` to `getOrAdd()`.  One thing I am not 100% sure is if we add `@FFDCIgnore()` here and this may throw `IllegalArgumentException` at the end and what is going to happen.  @donbourne ?

Reply:
That won't affect the program flow, it just affects whether or not you get FFDC instrumentation on your catch block.